partcad-client/tests/test_selfupdate.py

```
@@ -140,23 +140,199 @@ def test_repository_honors_the_environment(monkeypatch):
+    # host says so by leaving the level out, and that is reported against the
+    # release rather than against the manifest.
+    assert _select("linux", "x86_64", "ubuntu-24.04", manifest={"bundle": {}}) == []
+    assert _select("linux", "x86_64", "ubuntu-24.04", manifest={"bundle": {"linux": {}}}) == []
+
+
+def test_a_platform_list_that_is_a_string_is_rejected():
+    # Not filtered down to nothing and not iterated: every character of
+    # "ubuntu-24.04-x86_64" is a non-empty string, so a lenient reader would go
+    # looking for an archive named "u".
+    manifest = {"bundle": {"linux": {"x86_64": "ubuntu-24.04-x86_64"}}}
+    with pytest.raises(selfupdate.SelfUpdateError, match="bundle.linux.x86_64 is str, not a list"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)